compose/service.py

```diff
@@ -937,7 +938,19 @@ def build_volume_binding(volume_spec):
 
 
 def parse_volume_spec(volume_config):
+    """
+    A volume_config string, which is a path, split it into external:internal[:mode]
+    parts to be returned as a valid VolumeSpec tuple.
+    """
     parts = volume_config.split(':')
+
+    if IS_WINDOWS_PLATFORM:
+        # relative paths in windows expand to include the drive, eg C:\
+        # so we join the first 2 parts back together to count as one
+        windows_parts = [":".join(parts[0:2])]
+        windows_parts.extend(parts[2:])
+        parts = windows_parts
```

Comment:
I'd rather do that as a separate PR to this. This is making it work first without too big a change, then we can refactor and improve. I have a feeling there'll be some refactoring required with the other `volumes_from` PR as well.